rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -19,164 +19,332 @@
 import co.rsk.peg.federation.constants.FederationConstants;
 import co.rsk.peg.federation.constants.FederationMainNetConstants;
-import co.rsk.peg.federation.constants.FederationTestNetConstants;
 import java.time.Instant;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Optional;
 import java.util.stream.Stream;
 import co.rsk.peg.storage.StorageAccessor;
 import co.rsk.test.builders.FederationSupportBuilder;
 import org.bouncycastle.util.encoders.Hex;
 import org.ethereum.config.blockchain.upgrades.ActivationConfig;
 import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
+import org.ethereum.config.blockchain.upgrades.ConsensusRule;
 import org.ethereum.core.Block;
 import org.ethereum.crypto.ECKey;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.*;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
 
 class FederationSupportImplTest {
 
-    private FederationSupport federationSupport;
-    private FederationConstants federationMainnetConstants;
-    private NetworkParameters networkParameters;
+    private static final FederationConstants federationMainnetConstants = FederationMainNetConstants.getInstance();
+    private final Federation genesisFederation = FederationTestUtils.getGenesisFederation(federationMainnetConstants);
+    private Federation oldFederation;
+    private Federation newFederation;
     private FederationStorageProvider storageProvider;
-    private FederationSupportBuilder federationSupportBuilder;
+    private FederationSupportBuilder federationSupportBuilder = new FederationSupportBuilder();
+    private FederationSupport federationSupport;
 
     @BeforeEach
     void setUp() {
         StorageAccessor storageAccessor = new InMemoryStorage();
         storageProvider = new FederationStorageProviderImpl(storageAccessor);
-        federationMainnetConstants = FederationMainNetConstants.getInstance();
-        networkParameters = federationMainnetConstants.getBtcParams();
-        federationSupportBuilder = new FederationSupportBuilder();
 
         federationSupport = federationSupportBuilder
             .withFederationConstants(federationMainnetConstants)
             .withFederationStorageProvider(storageProvider)
             .build();
     }
 
-    @Test
-    void getActiveFederation_withNullFederations_returnsGenesisFederation() {
-        Federation genesisFederation = FederationTestUtils.getGenesisFederation(federationMainnetConstants);
-        Federation activeFederation = federationSupport.getActiveFederation();
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    @Tag("getActiveFederationTestsWithNullFederations")
+    class ActiveFederationTestsWithNullFederations {
+        @Tag("getActiveFederationTestsWithNullFederations")
+
+        @BeforeAll
```

Comment:
in each `TestInstance` we have to set the variables we wanna use, since [this](https://github.com/rsksmart/rskj/pull/2469/files#diff-835afc919a7b1b320019bf2bbba298e41cb0a9f11b4e60662b034be2762f683bR63) setUp is not being performed before.